File: src/main/java/org/revcloud/vader/types/validators/Validator.java
package org.revcloud.vader.types.validators;

import io.vavr.CheckedFunction1;
import io.vavr.control.Either;

/**
 * Data type for Validation functions.
 * This acts like a TypeAlias for {@link CheckedFunction1} 
 *
 *  @author gakshintala
 *  @since 228
 */
@FunctionalInterface
public interface Validator<ValidatableT, FailureT> extends CheckedFunction1<Either<FailureT, ValidatableT>, Either<FailureT, ?>> {
}

File: src/main/java/org/revcloud/vader/types/validators/Validator.kt
package org.revcloud.vader.types.validators

import io.vavr.CheckedFunction1
import io.vavr.control.Either

fun interface Validator<ValidatableT, FailureT> :
    CheckedFunction1<Either<FailureT, ValidatableT>, Either<FailureT, out Any?>> {}
